core/sql/cli/memorymonitor.cpp

```diff
@@ -105,14 +105,32 @@ MemoryMonitor::MemoryMonitor(Lng32 windowSize,
   char *currPtr;
   size_t bytesRead;
   fd_meminfo_ = fopen("/proc/meminfo", "r");
-  bytesRead = fread(buffer, 1, 1024, fd_meminfo_);
-  currPtr = strstr(buffer, "MemTotal");
-  sscanf(currPtr, "%*s " PF64 " kB", &memTotal_);
-  physKBytes_ = memTotal_;
-  physKBytesRatio_ = physKBytes_ / (8 * 1024 * 1024);
-  fd_vmstat_ = fopen("/proc/vmstat", "r");
+  if (fd_meminfo_) {
+    bytesRead = fread(buffer, 1, 1024, fd_meminfo_);
+    currPtr = strstr(buffer, "MemTotal");
+    if (currPtr) {
+      sscanf(currPtr, "%*s " PF64 " kB", &memTotal_);
+      physKBytes_ = memTotal_;
+      physKBytesRatio_ = physKBytes_ / (8 * 1024 * 1024);
+    }
+    else {
+      // something unexpected. let's just close the file
+      fclose(fd_meminfo_);
+      fd_meminfo_ = 0;
+    }
+  }
```

Comment:
physKBytesRatio_ doesn't seem to have an initializer so it will be uninitialized in some code paths. Could we add one?